cloud-minecraft-extras/src/main/java/org/incendo/cloud/minecraft/extras/parser/ComponentParserContributor.java

```diff
@@ -0,0 +1,106 @@
+                final boolean downsampleColors = annotation.downsampleColors();
+                final Function<String, ? extends Component> decoder =
+                    downsampleColors ? GsonComponentSerializer.colorDownsamplingGson()::deserialize
+                        : GsonComponentSerializer.gson()::deserialize;
+                return ParserParameters.single(MinecraftExtraParserParameters.COMPONENT_DECODER, decoder);
+            });
+            registry.registerAnnotationMapper(Decoder.class, (annotation, parsedType) -> {
+                final Function<String, ? extends Component> decoder;
+                try {
+                    decoder = annotation.value().getConstructor().newInstance().decoder(parsedType);
+                } catch (final ReflectiveOperationException exception) {
+                    throw new IllegalArgumentException("Could not create decoder for " + annotation.value(), exception);
+                }
+                return ParserParameters.single(MinecraftExtraParserParameters.COMPONENT_DECODER, decoder);
+            });
+
+            registry.registerParserSupplier(TypeToken.get(Component.class), options -> {
+                final boolean greedy = options.get(StandardParameters.GREEDY, false);
+                final boolean greedyFlagAware = options.get(StandardParameters.FLAG_YIELDING, false);
+                final boolean quoted = options.get(StandardParameters.QUOTED, false);
+                if (greedyFlagAware && quoted) {
+                    throw new IllegalArgumentException(
+                        "Don't know whether to create GREEDY_FLAG_YIELDING or QUOTED StringArgument.StringParser, both specified."
+                    );
+                } else if (greedy && quoted) {
+                    throw new IllegalArgumentException(
+                        "Don't know whether to create GREEDY or QUOTED StringArgument.StringParser, both specified."
+                    );
+                }
+                final StringParser.StringMode stringMode;
+                // allow @Greedy and @FlagYielding to both be true, give flag yielding priority
+                if (greedyFlagAware) {
+                    stringMode = StringParser.StringMode.GREEDY_FLAG_YIELDING;
+                } else if (greedy) {
+                    stringMode = StringParser.StringMode.GREEDY;
+                } else if (quoted) {
+                    stringMode = StringParser.StringMode.QUOTED;
+                } else {
+                    stringMode = StringParser.StringMode.SINGLE;
+                }
+                final Function<String, ? extends Component> decoder =
+                    options.get(MinecraftExtraParserParameters.COMPONENT_DECODER, MiniMessage.miniMessage()::deserialize);
+                return new ComponentParser<>(decoder, stringMode);
+            });
+        } catch (final Exception | LinkageError ignore) {
+            // ignore
```

Comment:
when would this happen? the lambdas shouldn't run unless the user uses the annotation

Reply:
I was just following the pattern of the text color parser contributor

Reply:
Oh I remember, it's for when extras happens to be on the classpath but adventure isn't... should have left a better comment